### ISSUE TITLE
the value of the residual vector quantization commitment loss

### ISSUE THREAD

**Issue by zjj1333** [facebookresearch/encodec#20]
## ❓ Questions
Hi， what is the value of the residual vector quantization commitment loss? The paper does not give the exact size of this value.
(Please ask your question here.)
![loss](https://user-images.githubusercontent.com/38657581/204696901-96794bfa-7a5f-4d07-88dd-6f05fb7f03ee.png)
![loss1](https://user-images.githubusercontent.com/38657581/204697119-8e5d4b00-8c14-43f7-ac9a-86c5f35d9f31.png)

**Comment by adefossez** [comment]
Good catch, we forgot to document it. The value is 1. Because the commitment loss is not defined with respect to the output of the model, it is not included in the balancer, and its gradient is computed separately and added to the one obtained from the balancer.

**Comment by Ximoo123** [comment]
> 
Hi, how to add the gradient of vector quantization commitment loss into the balancer? I try to use backpropagation to calculate the gradient value of vector quantization commitment loss separately, and then use the balancer to calculate the gradient of the remaining losses and update the network parameters as follows：
`loss_vq.backward(retain_graph = True)`
`balancer.backward(loss_g,x_hat)`
But I do not get good results.  Could you please share some methods to add this gradient to balancer?

**Comment by adefossez** [comment]
We do not add this to the balancer, the loss of the commitment loss is added during a second backward.
